CIPs/cip-draftultrageen.md

```diff
@@ -0,0 +1,69 @@
+---
+cip: <to be assigned>
+title: Ultragreen Celo
+author: <a list of the author's or authors' name(s) and/or username(s), or name(s) and email(s), e.g. (use with the parentheses or triangular brackets): FirstName LastName (@GitHubUsername), FirstName LastName <foo@bar.com>, FirstName (@GitHubUsername) and GitHubUsername (@GitHubUsername)>
+discussions-to: <URL>
+status: Draft
+type: Standards Track
+category (*only required for Standards Track): <Ring 0, 1, 2, 3>
+created: <date created on, in ISO 8601 (yyyy-mm-dd) format>
+license: Apache 2.0
+---
+
+## Simple Summary
+Celo has since its inception, in April 2020, had a focus on creating a positive impact on the plannet, becoming the first carbon-negative blockchain. This proposals wants to take that efford further and redistribute half of base fees of transactions to a new Green Fund (GF) and burn the other half, base fees are now sent fully to the Community Fund. The on-chain [Carbon Offsetting Fund](https://docs.celo.org/protocol/pos/epoch-rewards-carbon-offsetting-fund) also aims to be overhauled for a more transparent approach.
+
+## Abstract
+This proposals aims to oultine the requirements to make Celo fund on-chain ReFi initiatives with half of the base fees collected from transactions. The other half aims to improve the already deficitary tokenomics by burning it. The Carbon Offsetting Fund is meant to be transfered to an on-chain contract owned by Governance.
+
+## Motivation
+The motivation is to make Celo more efficient in how tokens are allocated and to bring more on-chain transparency about how the Carbon Offsetting Fund is run.
+
+## Specification
+
+### Current distribution of the base tx fees
+
+100% of the base fees are getting allocated to the Community Fund (Governance Contract Address).
+
+### Current distribution of Epoch Rewards
+
+Thought the on-chain [Carbon Offsetting Fund](https://docs.celo.org/protocol/pos/epoch-rewards-carbon-offsetting-fund), the blockchain is allocating 0.1% of the daily epoch rewards. 
+
+### Proposed distribution for base tx fees
+
+* Re-routing 50% of the base fees in Celo to the address `0x000000000000000000000000000000000000dEaD`. This will in practice take the Celo out of circulation. The other half will be sent to the CF.
+* 50% of the base fees paid in non-Celo tokens (currently Mento Token) will be send to a smart contract. That smart contract will use this tokens to buy Celo from on-chain markets and burn it. Examples of suitable on-chain markets are Mento, Uniswap and Ubeswap.
```

Comment:
What would happen in the case of low liquidity in the market? Or do we assume that the somewhat predictable demand will keep the liquidity high enough?

Reply:
In case it is not possible to trade to meet a given slippage requirement, then the transaction would revert, thus exchange of that token wouldn't be possible until more liquidity comes.